cpp/src/arrow_chunk_writer.cc

```diff
@@ -175,16 +190,111 @@ Status VertexPropertyWriter::WriteTable(
 Status EdgeChunkWriter::Validate(
     const std::shared_ptr<arrow::Table>& input_table, IdType vertex_chunk_index,
     ValidateLevel validate_level) const noexcept {
+  // use the writer's validate level
+  if (validate_level == ValidateLevel::default_validate)
+    validate_level = validate_level_;
+  // no validate
+  if (validate_level == ValidateLevel::no_validate)
+    return Status::OK();
+  // weak validate
+  if (!edge_info_.ContainAdjList(adj_list_type_))
+    return Status::InvalidOperation("invalid adj list type");
+  if (adj_list_type_ != AdjListType::ordered_by_source &&
+      adj_list_type_ != AdjListType::ordered_by_dest)
+    return Status::InvalidOperation("invalid adj list type");
```

Comment:
Ditto, "The adj list type has to be ordered_by_source or ordered_by_dest, but got xxx"